lib/pure/collections/sequtils.nim

```diff
@@ -675,19 +668,27 @@ template mapIt*(s, op: untyped): untyped =
     block:
       var it{.inject.}: type(items(s));
       op))
-  var result: seq[outType]
   when compiles(s.len):
-    evalOnce(t, s)
-    var i = 0
-    result = newSeq[outType](t.len)
-    for it {.inject.} in t:
-      result[i] = op
-      i += 1
+    block: # using a block avoids https://github.com/nim-lang/Nim/issues/8580
+
+      # avoid double substitution of `s`.
+      # NOTE: if we used `type(s) isnot openArray` instead, the openarray
+      # literal would be substituted twice
+      # BUG: `evalOnce(t, s, false)` would lead to C compile errors
+      # (`error: use of undeclared identifier`) instead of Nim compile errors
+      evalOnce(t, s, compiles((let _ = s)))
```

Comment:
``t``, ``s``

I know that you didn't introduce these names but I really despise single character variable names. In reviewing this PR I had to go back and forth between various templates to work out what `t` is meant to be here.

Perhaps they could be changed?

Reply:
Side note: this ``evalOnce`` macro feels like a total hack to me.

Reply:
almost procs in this module use single char arg names so I don't wanna special case mapIt, feel free to change that in a subsequent PR.
change s to s2 to make it obvious it relates to s; also improved var names of evalOnce and added a bunch of documentation.

> Side note: this evalOnce macro feels like a total hack to me.

feel free to improve that in a subsequent PR. It's there for a reason (avoid double evaluation bugs, handle openArray and other edge cases) and neither @LemonBoy nor I could come up with something better